web/src/main/webapp/js/graphhopper/GHRequest.js

```diff
@@ -201,10 +202,22 @@ GHRequest.prototype.createPath = function (url, skipParameters) {
         if(skipParameters && skipParameters[key])
             continue;
 
-        if (GHRoute.isArray(val)) {
-            for (var keyIndex in val) {
-                url += "&" + encodeURIComponent(key) + "=" + encodeURIComponent(val[keyIndex]);
+        if(typeof val === 'object') {
+            if (GHRoute.isArray(val)) {
+                var arr = val;
+                for (var keyIndex in arr) {
+                    var objKey = arr[keyIndex];
+                    url += "&" + encodeURIComponent(key) + "=" + encodeURIComponent(objKey);
+                }
+            } else {
+                // currently on one depth supported
+                var arr = Object.keys(val);
+                for (var keyIndex in arr) {
+                    var objKey = arr[keyIndex];
+                    url += "&" + encodeURIComponent(key) + "." + encodeURIComponent(objKey) + "=" + val[objKey];
+                }
```

Comment:
This is ugly and limited to one depth. The intention is clear: use properties with namespaces like `pt.` in the URL like we already do in the config. This will make the transition to JSON POST requests easier.